backend/app/views/spree/admin/orders/_shipment.html.erb

```diff
@@ -1,123 +1,121 @@
+        <tr class="edit-method hidden total">
           <td colspan="5">
-            <label><%= Spree.t(:tracking_number) %>:</label>
-            <%= text_field_tag :tracking, shipment.tracking %>
+            <div class="field alpha five columns">
+              <%= label_tag 'selected_shipping_rate_id', Spree::ShippingMethod.model_name.human %>
+              <%= select_tag :selected_shipping_rate_id,
+                    options_for_select(shipment.shipping_rates.map {|sr| ["#{sr.name} #{sr.display_price}", sr.id] }, shipment.selected_shipping_rate_id),
+                    {:class => 'select2 fullwidth', :data => {'shipment-number' => shipment.number } } %>
+            </div>
           </td>
           <td class="actions">
             <% if can? :update, shipment %>
-              <%= link_to '', '#', :class => 'save-tracking fa fa-check no-text with-tip', :data => {'shipment-number' => shipment.number, :action => 'save'}, :title => Spree.t('actions.save') %>
-              <%= link_to '', '#', :class => 'cancel-tracking fa fa-cancel no-text with-tip', :data => {:action => 'cancel'}, :title => Spree.t('actions.cancel') %>
+              <%= link_to '', '#', :class => 'save-method fa fa-check no-text with-tip',
+                :data => {'shipment-number' => shipment.number, :action => 'save'}, title: Spree.t('actions.save') %>
+              <%= link_to '', '#', :class => 'cancel-method fa fa-cancel no-text with-tip',
+                :data => {:action => 'cancel'}, :title => Spree.t('actions.cancel') %>
             <% end %>
           </td>
         </tr>
+        <% end %>
 
-        <% if order.special_instructions.present? %>
-          <tr class='special_instructions'>
-            <td colspan="5">
-              <strong><%= Spree.t(:special_instructions) %>:&nbsp;</strong><%= order.special_instructions %>
+        <tr class="show-method total">
+          <% if rate = shipment.selected_shipping_rate %>
+            <td colspan="4">
+              <strong><%= rate.name %></strong>
             </td>
-          </tr>
-        <% end %>
+            <td class="total" align="center">
+              <span><%= shipment.display_cost %></span>
+            </td>
+          <% else %>
+            <td colspan='5'><%= Spree.t(:no_shipping_method_selected) %></td>
+          <% end %>
 
-        <tr class="show-tracking total">
-          <td colspan="5" class="tracking-value">
-            <% if shipment.tracking.present? %>
-              <strong><%= Spree.t(:tracking) %>:</strong> <%= shipment.tracking %>
-            <% else %>
-              <%= Spree.t(:no_tracking_present) %>
-            <% end %>
-          </td>
           <td class="actions">
-            <% if can? :update, shipment %>
-              <%= link_to '', '#', :class => 'edit-tracking fa fa-edit no-text with-tip', :data => {:action => 'edit'}, :title => Spree.t('actions.edit') %>
+            <% if( (can? :update, shipment) and !shipment.shipped?) %>
```

Comment:
This line actually wasn't changed here, but looks like it because github's diff is confused by the massive whitespace change.

I've made this change in a second commit.